packages/fuels-core/src/code_gen/abigen/bindings/utils.rs

```diff
@@ -0,0 +1,71 @@
+use fuels_types::errors::Error;
+
+use crate::code_gen::abi_types::FullABIFunction;
+
+pub(crate) fn extract_main_fn(abi: &[FullABIFunction]) -> Result<&FullABIFunction, Error> {
+    let candidates = abi
+        .iter()
+        .filter(|function| function.name() == "main")
+        .collect::<Vec<_>>();
+
+    match candidates.as_slice() {
+        [single_main_fn] => Ok(single_main_fn),
+        _ => {
+            let fn_names = abi
+                .iter()
+                .map(|candidate| candidate.name())
+                .collect::<Vec<_>>();
+            Err(Error::CompilationError(format!(
+                "ABI must have one and only one function with the name 'main'. Got: {fn_names:?}"
+            )))
+        }
+    }
+}
+
```

Comment:
Don't know where else to put it. Don't feel like pulling it higher toward the mod file `bindings.rs` since it is only used by this lower level of the `bindings` mod. `bindings.rs` contains only what is needed to provide the bindings-generating functionality without getting bogged down in the details.

Can't inline it since it is used in two files, don't feel like merging the files since they handle scripts and predicates respectfully.

Guess it's a fear of having another `fuels-core/lib.rs` :D

Reply:
Oh I thought it was used only in one file, nevermind then